openupgrade_scripts/scripts/l10n_vn/17.0.2.0.3/pre-migration.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2024 Viindoo Technology Joint Stock Company (Viindoo)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_account_tax_group_xmlid = [
+    "l10n_vn.tax_group_0",
+    "l10n_vn.tax_group_10",
+    "l10n_vn.tax_group_5",
+]
+
+_account_tax_xmlid = [
+    "l10n_vn.tax_purchase_vat0",
+    "l10n_vn.tax_purchase_vat10",
+    "l10n_vn.tax_purchase_vat5",
+    "l10n_vn.tax_sale_vat0",
+    "l10n_vn.tax_sale_vat10",
+    "l10n_vn.tax_sale_vat5",
+]
+
+
+def _vn_coa_rename_xml_id(env):
+    """
+    Since the removal of account.chart.template
+    we need to rename some xml_id like tax or tax.group
+    in order to avoid duplication
+    """
+    env.cr.execute("""SELECT id, name FROM res_company WHERE chart_template = 'vn'""")
+    xmlids_renames = []
+    for company_id, _ in env.cr.fetchall():
```

Comment:
name không dùng thì không cần select làm gì
```suggestion
    env.cr.execute("""SELECT id FROM res_company WHERE chart_template = 'vn'""")
    xmlids_renames = []
    for company_id in env.cr.fetchall():
```